cmd/skopeo/tags_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type tagListing struct {
+	DockerRepository string
+	Tags             []string
+}
+
+func TestListTags(t *testing.T) {
+	// Invalid command-line arguments
+	for _, args := range [][]string{
+		{},
+		{"invalid1", "invalid2"},
+	} {
+		out, err := runSkopeo(append([]string{"docker-list-tags"}, args...)...)
+		assertTestFailed(t, out, err, "Usage:")
+	}
+
+	// invalid repository name
+	out, err := runSkopeo("docker-list-tags", "/this/doesnt/exist")
+	assertTestFailed(t, out, err, "invalid reference format")
```

Comment:
The above is great to have here; I’m not sure that the tests below, which actually verify run-time behavior against a registry, are a good fit for unit tests.

Running them in the integration test phase might be a better fit, both against `docker.io` or possibly against the test registry with precisely known contents (i.e. the tests could also include correctness of the output.)

Reply:
Yes agree. I'll remove from unit-test and explore the integration tests.